ambari-server/src/main/resources/stack-hooks/after-INSTALL/scripts/shared_initialization.py

```diff
@@ -84,7 +84,7 @@ def setup_config():
     XmlConfig("core-site.xml",
               conf_dir=params.hadoop_conf_dir,
               configurations=params.module_configs.get_property_value(params.module_name, 'core-site', ''),
```

Comment:
This is wrong. 
This should be module_configs.get_all_properties(params.module_name, 'core-site') which should return all core-site properties.

Reply:
Now there is no get_all_properties() in execution_command module. Of course I can add it. However, there it seems different from your comment. Based on my understand, the configurations return from this call is a dict object:

original code:
configurations=params.config['configurations']['core-site']
If we use "module_configs.get_all_properties(params.module_name, 'core-site') which should return all core-site properties.", does that mean the return is a list of attributes?

Reply:
The updated api in module_configs:

  def get_properties(self, module_name, config_type, property_names):
    return map(lambda property_name: self.get_property_value(module_name, config_type, property_name), property_names)

  def get_property_value(self, module_name, config_type, property_name, default=None):
    try:
      if property_name:
        return self.__module_configs[config_type][property_name]
      else:
        return self.__module_configs[config_type]
    except:
      return default

Reply:
How can get_property_value() API return values for all properties for that config type. The method is for retrieving a single property value but also returns all property values?